### ISSUE TITLE
RF02 triggered when referring table by alias in a BROADCAST hint

### ISSUE THREAD

**Issue by TiagoSilva67** [sqlfluff/sqlfluff#6771]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When referring a table by it's alias inside a BROADCAST hint, RF02 is reported

### Expected Behaviour

RF02 should not be reported, as the table does not need a qualifier

### Observed Behaviour

L: 1| P:  20 | RF02 | Unqualified reference 'systems' found in select with more than one
                       | referenced table/view. [references.qualification]

### How to reproduce

```sql
select /*+ broadcast(systems) */
        source.key,
        source.a,
        source.b,
        othertable.x
from source_table as source
join other_systems as systems on
    source.key = systems .key
```

### Dialect

sparksql

### Version

3.3.0

### Configuration

[sqlfluff]
# Level of log output
verbose = 0
# Turn off color formatting of output
nocolor = False
# SQL dialect to lint
dialect = sparksql
# Templating best supported in Terraform parsing
templater = placeholder
# Max line length for SQL files
max_line_length = 200
# SQLFluff output length
output_line_length = 100
# Number of passes to run before admitting defeat
runaway_limit = 10
# Ignore parsing errors due to some keywords not being detected as keywords(eg. sort, distribute, merge)
ignore = PRS
# Ignore inline overrides (e.g. to test if still required)
disable_noqa = False
# Whether to warn about unneeded '-- noqa:' comments.
warn_unused_ignores = False
# Ignore linting errors found in code from templated code
ignore_templated_areas = True
# Encoding of the files to lint
encoding = autodetect
# List of file extensions to lint
sql_file_exts = .sql,.sql.j2,.dml,.ddl
# Allow fix to run on files, even if they contain parsing errors
fix_even_unparsable = False
# Disabled
large_file_skip_char_limit = 0
# Disabled
large_file_skip_byte_limit = 0
# Number of processes to use while linting
processes = 1
# List of rules to check
rules = layout, AL01, AL02, AL04, AL05, AL08, AM01, AM03, AM07, CP01, CP03, CP04, CP05, CV01, CV02, CV03, CV10, CV11, CV12, JJ01, LT01, LT02, LT03, LT04, RF02, RF04, RF05, RF06, ST02, ST03, ST04, ST06, ST07, ST09, ST11
# Non enforced rules displayed as warnings
# Current warnings:
# AM06: Inconsistent column references in GROUP BY/ORDER BY clauses
# CP02: Unquoted identifiers must be lower case
# LT01: Inappropriate Spacing
# LT02: Incorrect Indentation
# LT05: Line is too long
# LT09: Select targets should be on a new line unless there is only one select target
# LT12: Files must end with a single trailing newline.
# ST02: Unnecessary CASE statement
# ST05: Join/From clauses should not contain subqueries. Use CTEs instead
# ST06: Select wildcards then simple targets before calculations and aggregates
warnings = AM06, LT05, LT09, LT12, ST05
# List of rules to exclude
exclude_rules = LT08

[sqlfluff:templater:placeholder]
# Dollar sign identifies the placeholders
param_style = dollar

[sqlfluff:indentation]
# See https://docs.sqlfluff.com/en/stable/layout.html#configuring-indent-locations
indent_unit = space
tab_space_size = 4
indented_joins = False
indented_ctes = False
indented_using_on = False
indented_on_contents = True
indented_then = True
indented_then_contents = True
allow_implicit_indents = False
template_blocks_indent = True
# This is a comma separated list of elements to skip
skip_indentation_in = script_content
trailing_comments = before

##########################
##########################
# See https://docs.sqlfluff.com/en/stable/layout.html#configuring-layout-and-spacing

[sqlfluff:layout:type:comma]
spacing_before = touch
line_position = trailing

[sqlfluff:layout:type:binary_operator]
spacing_within = touch
line_position = leading

[sqlfluff:layout:type:statement_terminator]
spacing_before = touch
line_position = trailing

[sqlfluff:layout:type:end_of_file]
spacing_before = touch

[sqlfluff:layout:type:set_operator]
line_position = alone:strict

[sqlfluff:layout:type:start_bracket]
spacing_after = touch

[sqlfluff:layout:type:end_bracket]
spacing_before = touch

[sqlfluff:layout:type:start_square_bracket]
spacing_after = touch

[sqlfluff:layout:type:end_square_bracket]
spacing_before = touch

[sqlfluff:layout:type:start_angle_bracket]
spacing_after = touch

[sqlfluff:layout:type:end_angle_bracket]
spacing_before = touch

[sqlfluff:layout:type:casting_operator]
spacing_before = touch
spacing_after = touch:inline

[sqlfluff:layout:type:slice]
spacing_before = touch
spacing_after = touch

[sqlfluff:layout:type:comparison_operator]
spacing_within = touch
line_position = leading

[sqlfluff:layout:type:assignment_operator]
spacing_within = touch
line_position = leading

[sqlfluff:layout:type:object_reference]
spacing_within = touch:inline

[sqlfluff:layout:type:numeric_literal]
spacing_within = touch:inline

[sqlfluff:layout:type:sign_indicator]
spacing_after = touch:inline

[sqlfluff:layout:type:tilde]
spacing_after = touch:inline

[sqlfluff:layout:type:function_name]
spacing_within = touch:inline
spacing_after = touch:inline

[sqlfluff:layout:type:array_type]
spacing_within = touch:inline

[sqlfluff:layout:type:typed_array_literal]
spacing_within = touch

[sqlfluff:layout:type:sized_array_type]
spacing_within = touch

[sqlfluff:layout:type:struct_type]
spacing_within = touch:inline

[sqlfluff:layout:type:bracketed_arguments]
spacing_before = touch:inline

[sqlfluff:layout:type:typed_struct_literal]
spacing_within = touch

[sqlfluff:layout:type:semi_structured_expression]
spacing_within = touch:inline
spacing_before = touch:inline

[sqlfluff:layout:type:array_accessor]
spacing_before = touch:inline

[sqlfluff:layout:type:colon]
spacing_before = touch

[sqlfluff:layout:type:comment]
spacing_before = any
spacing_after = any

[sqlfluff:layout:type:placeholder]
# Placeholders exist "outside" the rendered SQL syntax
# so we shouldn't enforce any particular spacing around
# them.
spacing_before = any
spacing_after = any

[sqlfluff:layout:type:common_table_expression]
# The definition part of a CTE should fit on one line where possible.
# For users which regularly define column names in their CTEs they
# may which to relax this config to just `single`.
spacing_within = single:inline

# By setting a selection of clauses to "alone", we hint to the reflow
# algorithm that in the case of a long single line statement, the
# first place to add newlines would be around these clauses.
# Setting this to "alone:strict" would always _force_ line breaks
# around them even if the line isn't too long.
[sqlfluff:layout:type:select_clause]
line_position = alone

[sqlfluff:layout:type:where_clause]
line_position = alone

[sqlfluff:layout:type:from_clause]
line_position = alone

[sqlfluff:layout:type:join_clause]
line_position = alone

[sqlfluff:layout:type:groupby_clause]
line_position = alone

[sqlfluff:layout:type:orderby_clause]
# NOTE: Order by clauses appear in many places other than in a select
# clause. To avoid unexpected behavior we use `leading` in this
# case rather than `alone`.
line_position = leading

[sqlfluff:layout:type:having_clause]
line_position = alone

[sqlfluff:layout:type:limit_clause]
line_position = alone

# Template loop tokens shouldn't dictate spacing around them.
[sqlfluff:layout:type:template_loop]
spacing_before = any
spacing_after = any

###########
###########

# General properties, applies to all rules
# Reference: https://docs.sqlfluff.com/en/stable/reference/rules.html#rule-index
[sqlfluff:rules]
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all

#####################
#####################

# Rule AL01 - Tables must include an explicit alias
[sqlfluff:rules:aliasing.table]
# Aliasing preference for tables
aliasing = explicit

# Rule AL02 - Columns must include an explicit alias
[sqlfluff:rules:aliasing.column]
# Aliasing preference for columns
aliasing = explicit

######################
######################

# Rule AM06 - References in GROUP BY/ORDER BY clauses must be consistent
[sqlfluff:rules:ambiguous.column_references]
group_by_and_order_by_style = consistent

###########################
###########################

# Rule CP01 - Keywords must be lowercase
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower
ignore_words = None
ignore_words_regex = None

# Rule CP02 - Unquoted identifiers must be lower case
[sqlfluff:rules:capitalisation.identifiers]
# Unquoted identifiers
extended_capitalisation_policy = lower
# Words that are in uppercase or camelcase in the source schemas should be ignored
ignore_words = KAFKA_DATA, PurchaseRequisition, SchedulingAgreement, LogisticSinglePurchaseOrder, MRPStockExclusion, IV_MANDT, IV_EXMOD, IV_PLSCN, IV_MATNR, IV_WERKS, IV_BERID, IV_PLAAB, IV_PLANR, IV_ERGBZ, IV_CONTENTTYPE, IV_CONTENTMETHOD, IV_TRANSACTIONID, IV_CREATEDTIMESTAMP, IV_SYSTEMID, IS_CM61B, IS_CM61M, IS_MDKP, IS_MDSTA, IS_MT61D, IS_ZDATA, IT_BMW_MDPSX, IT_MDPSX, IT_MDEZX, IT_MDSUX
ignore_words_regex = ['[a-z][A-Z]|[A-Z][a-z]']

# Rule CP03 - Functions must be lowercase
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower
ignore_words = None
ignore_words_regex = None

# Rule CP04 - Literals must be lowercase
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower
ignore_words = None
ignore_words_regex = None

# Rule CP05 - Types must be lowercase
[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower
ignore_words = None
ignore_words_regex = None

#######################
#######################

# Rule CV03 - Trailing commas in SELECT clauses are forbidden
[sqlfluff:rules:convention.select_trailing_comma]
select_clause_trailing_comma = forbid

# Rule CV10 - Quoted literals must be consistent
[sqlfluff:rules:convention.quoted_literals]
preferred_quoted_literal_style = consistent
# Disabled for dialects that do not support single and double quotes for quoted literals
force_enable = False

# Rule CV11 - Type casting functions must be consistent
[sqlfluff:rules:convention.casting_style]
preferred_type_casting_style = consistent

###################
###################

# Rule LT05 - Comment lines must respect max length
[sqlfluff:rules:layout.long_lines]
# Line length
ignore_comment_lines = False
ignore_comment_clauses = False

#######################
#######################

# Rule RF04 - Keywords should not be used as identifiers
[sqlfluff:rules:references.keywords]
# Ignore these keywords
ignore_words = distribute, sort, merge, date, language, trigger
# If unquoted, flag aliases
unquoted_identifiers_policy = aliases
# If quoted, flag nothing
quoted_identifiers_policy = none

# Rule RF05 - Do not allow special characters in identifiers
[sqlfluff:rules:references.special_chars]
additional_allowed_characters = None
allow_space_in_identifier = False
ignore_words = None
ignore_words_regex = None
quoted_identifiers_policy = all
unquoted_identifiers_policy = all

# Rule RF06 - Policy on quoted and unquoted identifiers
[sqlfluff:rules:references.quoting]
ignore_words = None
ignore_words_regex = None

######################
######################

# Rule ST05 - Use CTEs instead, subqueries in FROM clauses allowed. Join clauses forbidden.
[sqlfluff:rules:structure.subquery]
forbid_subquery_in = join

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
